### PR TITLE
Feature/user menu

### PR DESCRIPTION
**Pull request by MichaelCastiau**
## PR Checklist

This PR fulfills the following requirements:

- [ x] The commit message follows our guidelines: [Contributing guidelines](https://github.com/digipolisantwerp/acpaas-ui_angular/blob/master/CONTRIBUTING.md)
- [ x] Tests for the changes have been added (for bug fixes / features)
- [ x] Docs have been added / updated (for bug fixes / features)
- [ x] A [changelog entry](https://github.com/digipolisantwerp/acpaas-ui_angular/blob/master/guidelines/CHANGELOG.md) was added

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the new behavior?

UserMenuComponent is created, Local docker configuration is added for development purposes

## Does this PR introduce a breaking change?

- [ ] Yes
- [x ] No

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -70,13 +72,31 @@ $ npm run examples
 $ npm run styleguide
 ```
 
+## Using Docker
+
+Aditionally to running this project locally, the repository also offers you the option to develop using Docker containers.
```
Comment (jsebrech) [inline CONTRIBUTING.md]: Ad**d**itionally

File: packages/user-menu/examples/src/user-menu/pages/demo/demo.page.html [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+<h3 class="u-margin-bottom-xs">Demo</h3>
+
+<p class="u-text-bold u-margin-bottom">
+	The User Menu component can be used on a webpage to provide visual login and logout buttons.
+	If logged out, the component displays a single button enabling the user to login.
+	If logged in, the component provides an avatar button, displaying a flyout where a logout button is featured.
+</p>
+<p class="u-margin-bottom">
+	This component uses <strong>content projection</strong> inside it's tag, meaning whatever html is included
```
Comment (jsebrech) [inline packages/user-menu/examples/src/user-menu/pages/demo/demo.page.html]: its

File: packages/user-menu/src/lib/user-menu/components/user-menu/user-menu.component.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<div class="m-user-menu-container">
+	<!-- Container showing the login button -->
+	<ng-container *ngIf="!user">
+		<button class="a-button a-button-login has-icon-left" title="Aanmelden"
+			alt="Klik hier om u zich aan te melden met uw A-profiel" (click)="login$.next()">
```
Comment (jsebrech) [inline packages/user-menu/src/lib/user-menu/components/user-menu/user-menu.component.html]: "Klik hier om aan te melden met uw A-profiel"

File: packages/user-menu/src/lib/user-menu/components/user-menu/user-menu.component.scss [hunk 4]
```diff
@@ -0,0 +1,23 @@
+@import '~@a-ui/core/dist/assets/styles/_quarks';
+
+.a-button-avatar {
```
Comment (TriangleJuice) [inline packages/user-menu/src/lib/user-menu/components/user-menu/user-menu.component.scss]: A few remarks:

- there are some indenting issues
- please add spaces above new selectors
- make sure all colons are followed by a space
- please arrange your properties in alphabetical order (we try to do this in all projects)

File: packages/flyout/lib/src/flyout/types/flyout.types.ts [hunk 5]
```diff
@@ -10,3 +10,7 @@ export enum FlyoutState {
 	OPEN,
 	CLOSED,
 }
+
+export declare namespace Flyout {
+	export type EFlyoutSize = FlyoutSize;
```
Comment (TriangleJuice) [inline packages/flyout/lib/src/flyout/types/flyout.types.ts]: What's the `E` for?

File: scripts/helpers/deps.js [hunk 6]
```diff
@@ -43,8 +43,16 @@ module.exports.getNPMDependencies = ({ preserve = false } = {}) => {
 	const isDirectory = source => lstatSync(source).isDirectory();
 	const getDirectories = source => readdirSync(source).map(name => join(source, name)).filter(isDirectory);
 
-	const packages = getDirectories(resolve(process.cwd(), 'packages')).map(path => `${path}/lib/`);
-
+	const packages = getDirectories(resolve(process.cwd(), 'packages')).map(path => {
```
Comment (TriangleJuice) [inline scripts/helpers/deps.js]: Am I right that this code 'fixes' the wrong scaffold provided by our schematics? (part 2)

File: scripts/build.js [hunk 7]
```diff
@@ -9,9 +9,20 @@ const buildPackage = require('./helpers/build');
 const exec = require('./helpers/bash');
 
 const directories = getDirectories(resolve(process.cwd(), 'packages'));
-const configs = directories.map(directory => readFileSync(resolve(`${directory}/lib`, 'package.json'), {
-	encoding: 'UTF-8'
-}));
+const configs = directories.map(directory => {
```
Comment (TriangleJuice) [inline scripts/build.js]: Am I right that this code 'fixes' the wrong scaffold provided by our schematics? (part 1)

Comment (MichaelCastiau) [reply scripts/build.js]: Yep

File: scripts/helpers/deps.js (hunk 6 above)
Comment (MichaelCastiau) [reply scripts/helpers/deps.js]: Yep

File: packages/flyout/lib/src/flyout/types/flyout.types.ts (hunk 5 above)
Comment (MichaelCastiau) [reply packages/flyout/lib/src/flyout/types/flyout.types.ts]: Enum

**Pull request merged**